### ISSUE TITLE
Store inode when reading /proc/*/smaps in _pslinux.py/get_memory_maps()

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#444]
_From [remi.cha...@gmail.com](https://code.google.com/u/110855288675224986459/) on November 20, 2013 12:12:24_

```
At the moment, the function get_memory_maps() in the Linux implementation 
_pslinux.py, correctly parses the content of /proc/<pid>/smaps and stores some 
of the memory maps data in a container.

Unfortunately, it does not store the inode, although it is correctly parsed 
from first_line.

This is specifically necessary with SYSV00000000 maps, because their path names 
are not unique.

It would be great to store the inode as a new item in the returned container. 
It does not imply changes to existing applications.

On the other hand, I do not know how this can be emulated on other operating systems.
```

_Original issue: http://code.google.com/p/psutil/issues/detail?id=444_

**Comment by ganguin** [comment]
Is this still on track?

I'm developing a tool that track processes that are running with outdated mmap'ed libraries (containing bugs or security issues or even a malware infected file using this to hide). /proc/pid/smap flags them with appending " (deleted)" at the end of the path.

psutil does not allow to get the inode which would but required to validate the mmap'ed file is different from the file on the filesystem. psutil also strips the " (deleted)". Therefore I see no way of getting the info for my case.

like remi mentioned in the first comment the patch seems trivial:
* Line 1672 already reads the inode from /proc/_/smaps
https://github.com/giampaolo/psutil/blob/f0094db79ad9e4f2246997cb8c2046b71c465a29/psutil/_pslinux.py#L1672-L1674
* the inode number could be stored as a new item in the returned container
https://github.com/giampaolo/psutil/blob/f0094db79ad9e4f2246997cb8c2046b71c465a29/psutil/_pslinux.py#L1685-L1696

I don't know about other OS, however as I can see in the doc, the info is already very different depending on the OS:
http://psutil.readthedocs.io/en/latest/#psutil.Process.memory_maps

**Comment by giampaolo** [comment]
Yes, info is different depending on the OS, but all the values that differ are integers that are summed together when using `grouped=True` argument. Adding the inode would make this kinda twisted because we should exclude it from the `sum()` logic. 
You use case (identifying "dead" / "(deleted)" mmap files) looks very specific and wouldn't benefit from adding the inode alone (correct?).
